orchagent/tunneldecaporch.cpp

```diff
@@ -510,43 +624,45 @@ bool TunnelDecapOrch::setTunnelAttribute(string field, string value, sai_object_
  *
  * Arguments:
  *    @param[in] key - key of the tunnel from APP_DB
- *    @param[in] new_ip_addresses - new destination ip addresses to decap (comes from APP_DB)
+ *    @param[in] src_ip - source IP for tunnel term
+ *    @param[in] new_dst_ip_addresses - new destination ip addresses to decap (comes from APP_DB)
  *    @param[in] tunnel_id - the id of the tunnel
+ *    @param[in] term_type - The type of tunnel term
  *
  * Return Values:
  *    @return true on success and false if there's an error
  */
-bool TunnelDecapOrch::setIpAttribute(string key, IpAddresses new_ip_addresses, sai_object_id_t tunnel_id)
+bool TunnelDecapOrch::setIpAttribute(string key, IpAddress src_ip, IpAddresses new_dst_ip_addresses, sai_object_id_t tunnel_id, TunnelTermType term_type)
 {
     TunnelEntry *tunnel_info = &tunnelTable.find(key)->second;
 
     // make a copy of tunnel_term_info to loop through
     vector<TunnelTermEntry> tunnel_term_info_copy(tunnel_info->tunnel_term_info);
 
     tunnel_info->tunnel_term_info.clear();
-    tunnel_info->dst_ip_addrs = new_ip_addresses;
-
+    tunnel_info->dst_ip_addrs = new_dst_ip_addresses;
```

Comment:
Please have a section in the beginning for p2p and retain the previous code as-is